float64 mouth_x
float64 mouth_y
float64 dist 